### ISSUE TITLE
Angular 8 with JSDO

### ISSUE THREAD

**Issue by hutcj** [progress/JSDO#266]
Regarding the following JSDO for Angular node packages:
https://www.npmjs.com/package/@progress/jsdo-core
https://www.npmjs.com/package/@progress/jsdo-angular

We began using these packages in an Angular 7 project close to a year ago. We would always get warning messages that the JSDO packages required peer dependency of Angular version 6.1. However, since the code worked as expected, we ignored the warnings.

Now we would like to upgrade to Angular 8 since it is now available. However, ng update refuses to upgrade versions because JSDO has declared version 6.1 of Angular a peer dependency. When is this going to be updated?

The update process for Angular apps seems fairly straightforward: https://update.angular.io/#6.1:7.0

**Comment by zerovian** [comment]
Hi hutcj,

The JSDO is a community supported project and we welcome contributions.  Angular 8 support is in the backlog, but right now is not a high priority for the development team.  If you care to issue a pull request with the requisite changes needed, we'd be happy to review the work.

**Comment by hutcj** [comment]
Hi zerovian,

As far as I can tell, the version update would require nothing more than updating the @angular/core references to be something like `"@angular/core": "^6.1.0 - ^8.0.0"` instead of `"@angular/core": "^6.1.0"` as there are no relevant breaking changes between ng 6-7 and ng 7-8.

I have committed this change in a branch, following the guidelines posted here: https://github.com/progress/JSDO/blob/master/contributing.md#contribute-to-the-code-base

However, I got rejected when trying to push up my feature branch.
![image](https://user-images.githubusercontent.com/40180018/64450622-84a8fe80-d0b0-11e9-8353-4348361fee90.png)

**Comment by zerovian** [comment]
Hi hutcj,

sorry for the lag.  we're in the middle of PI planning so every one is ignoring some of their responsibilities...like me.  

you can't push directly to this repository.  You would need to be listed as a team member.  Normally the process is to clone the repository in github so you have your own personal copy of the repo, then make your changes in a  feature branch, based off the develop branch, in your own repo.  After you build and test, then issue a pull request to progress/JSDO upstream repository, and we'll review it and merge the pull request if it passes our tests.

**Comment by hutcj** [comment]
https://github.com/progress/JSDO/pull/267

**Comment by edselg** [comment]
Hello @hutcj 

Thank you for reporting this issue and submitting a pull request.

I wanted to share with you the results of my testing.

I tried to reproduce this issue by installing Angular 7.3.9 then trying to upgrade to 8.3.9 and in this way test your changes and determine if additional changes were required.

I used the following steps:
* ng -g install @angular/cli@7.3.9
* ng new jsdo-angular
cd jsdo-angular
* npm install @progress/jsdo-angular
** This install @progress/jsdo-core@6.0.1 and @progress/jsdo-angular@6.0.0

* ng serve

**Note:**
These steps can be found in the following wiki:
* [Using the JSDO and DataSource components in an Angular web app (Angular 6.x)](https://github.com/progress/JSDO/wiki/Using-the-JSDO-and-DataSource-components-in-an-Angular-web-app-(Angular-6.x))

I reproduced the issue by running "ng update" on the Angular project:
* ng update
* ng update --all

![image](https://user-images.githubusercontent.com/12529263/66837397-eb7fca00-ef30-11e9-9275-7ef375da2a16.png)

The issue goes away after updating @progress/jsdo-angular with a version (new 7.0.0) that includes your proposed fix of using @angular/core 8.0.0 as a peer dependency instead of 6.1.0.

![image](https://user-images.githubusercontent.com/12529263/66845605-4f5cbf80-ef3e-11e9-895f-c0bf5683524b.png)

**Note:** The Angular project may error out with the following message:
ERROR in The Angular Compiler requires TypeScript >=3.4.0 and <3.6.0 but 3.6.4 was found instead.

I installed TypeScript 3.4.5.

An additional change is required in @progress/jsdo-core to work with Angular 8:

```
    var _global = (0, eval)('this');

    if (typeof progress === "undefined") {
        _global.progress = {};
    }
```

The application runs fine using @progress/jsdo-angular@7.0.0 with Angular 7.0.0 as well as with Angular 8.0.0. The difference is that there would be no peer warning when using jsdo-angular 7.0.0 and Angular 8.0.0 together.
The application runs fine with anonymous, basic and form authentication.

![image](https://user-images.githubusercontent.com/12529263/66848720-9600e880-ef43-11e9-82a3-61366a20588c.png)

I will merge your pull request and perform the additional required changes.

Thank you and regards.

**Comment by hutcj** [comment]
Thanks, Edsel!

**Comment by hutcj** [comment]
@edselg I see this was merged, but is a new version going to be released soon with the changes?

**Comment by edselg** [comment]
@hutcj 

I have not heard of a new version being released soon.
You can mention of your need to your contact at Progress.

In the meantime, you can build your own packages from the develop branch.

Here are the instructions:

1. git clone https://github.com/progress/JSDO.git
2. cd JSDO
3. git checkout develop
4. npm install
5. npm run build:jsdo
6. npm run build:jsdo-core
7. npm run pack:jsdo-core
8. cd packages\ng-datasource
9. npm install ..\..\build\packages\progress-jsdo-core-7.0.0.tgz
10. npm install
11. npm install @angular/core@8.0.0
12. npm install rxjs@6.5.3
13. npm install zone.js@~0.9.1
14. npm install @types/node@8
15. npm run build:ds
16. npm run build:jsdo-angular
17. npm run pack:jsdo-angular

Once you have build the packages for jsdo-core and jsdo-angular, then you install the new version in your Angular project:
* npm install D:\work\jsdo\build\packages\progress-jsdo-core-7.0.0.tgz
* npm install D:\work\jsdo\packages\ng-datasource\build\packages\progress-jsdo-angular-7.0.0.tgz
* ng serve

I hope this helps.

Please let me know how it goes.

Thank you and regards.

**Comment by zerovian** [comment]
@hutcj  there are no short term plans to release a new version of JSDO.  Next program increment starts end of January and discussions are ongoing as to what work will be done.  Best bet, for now, is to clone and build it locally following @edselg  instructions.

**Comment by edselg** [comment]
Issue was fixed via pull request #269 .
